src/Parser.php

```diff
@@ -35,107 +38,67 @@ public function __construct()
     }
 
     /**
-     * @param string|resource $input File path or resource
+     * @param string|resource   $input        File path or resource
      * @param callback|callable $itemCallback Callback
-     * @param bool $assoc Parse as associative arrays
+     * @param bool              $assoc        Parse as associative arrays
      *
-     * @throws \Exception
+     * @throws Exception
      */
-    public function parse($input, $itemCallback, $assoc = true)
+    public function parse($input, $itemCallback, bool $assoc = true): void
     {
         $this->checkCallback($itemCallback);
 
-        $stream = $this->openStream($input);
-
-        try {
-            $listener = new Listener($itemCallback, $assoc);
-            $this->parser = new BaseParser(
-                $stream,
-                $listener,
-                $this->getOption('line_ending'),
-                $this->getOption('emit_whitespace')
-            );
-            $this->parser->parse();
-        } catch (\Exception $e) {
-            $this->closeStream($stream);
-            throw $e;
-        }
-
-        $this->closeStream($stream);
+        $this->parseStream($input, new Listener($itemCallback, $assoc));
     }
 
     /**
-     * @param string|resource $input File path or resource
+     * @param int               $size         Size of the chunk to collect before processing
+     * @param string|resource   $input        File path or resource
      * @param callback|callable $itemCallback Callback
+     * @param bool              $assoc        Parse as associative arrays
      *
-     * @throws \Exception
+     * @throws Exception
      */
-    public function parseAsObjects($input, $itemCallback)
+    public function chunk(int $size, $input, $itemCallback, bool $assoc = true): void
```

Comment:
@jasonhebert any strong argument why are you putting $size as the first argument? Looks like not very consistent with ->parse method.

Reply:
none, will modify

Reply:
Corrected in latest commit